yt_dlp/extractor/afl.py

```diff
@@ -0,0 +1,204 @@
+    }, {
+        'url': 'https://www.carltonfc.com.au/podcasts/summer-sessions',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        element = get_element_by_class('omny-embed', webpage)
+        podcast_url = traverse_obj(extract_attributes(element), ('src', {url_or_none}))
+        return self.url_result(podcast_url, OmnyFMShowIE)
+
+
+class AFCVideoIE(InfoExtractor):
+    IE_NAME = 'afc:video'
+    _VALID_URL = r'https?://(?:www\.)?afc\.com.au/video/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://www.afc.com.au/video/1657583/girls-academies-be-a-pro?videoId=1657583&modal=true&type=video&publishFrom=1726548621001',
+        'md5': 'd0f4ec78b5a693d95c975ae3aeed8b2d',
+        'info_dict': {
+            'id': '6362048189112',
+            'ext': 'mp4',
+            'description': 'md5:5c43f1affe1a0cd8e2192358a49de9cc',
+            'upload_date': '20240917',
+            'duration': 50.48,
+            'tags': 'count:0',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'title': 'Girls Academies – ‘Be a Pro’',
+            'uploader_id': '6057984922001',
+            'timestamp': 1726548942,
+        },
+    }, {
+        'url': 'https://www.afc.com.au/video/1586280/se10ep16-the-crows-show?videoId=1586280&modal=true&type=video&publishFrom=1719639000001&tagNames=crowsshowepisode',
+        'md5': 'bd9984d62f87b4c2299bb62ffc869189',
+        'info_dict': {
+            'id': '6355746458112',
+            'ext': 'mp4',
+            'description': 'md5:4470d107af6e749a8225fd558b98b50b',
+            'upload_date': '20240627',
+            'duration': 1193.64,
+            'tags': 'count:0',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'title': 'SE10EP16 - The Crows Show',
+            'uploader_id': '6057984922001',
+            'timestamp': 1719466601,
+        },
+    }, {
+        'url': 'https://www.afc.com.au/video/1634706/jones-radiology-injury-update-r24?videoId=1634706&modal=true&type=video&publishFrom=1724126172001',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        video_attrs = extract_attributes(get_element_html_by_id('VideoModal', webpage))
+        player_id = video_attrs['data-player-id'] + '_default'
+        account_id = video_attrs['data-account-id']
+
+        video_element_html = get_element_html_by_attribute('data-id', display_id, webpage)
+        if video_element_html is None:
+            data = self._download_json(f'https://aflapi.afc.com.au/content/aflc-adel/video/en/{display_id}', display_id)
+            video_id = traverse_obj(data, ('mediaId', {str_or_none}))
+        else:
+            video_id = self._search_regex(r'"mediaId"\s*:\s*"(\d+)"', video_element_html, 'video-id', fatal=False)
```

Comment:
Is `mediaId` an attribute of the element you extracted beforehand?

```suggestion
        video_element_html = get_element_html_by_attribute('data-id', display_id, webpage)
        if not video_element_html:
            video_data = self._download_json(f'https://aflapi.afc.com.au/content/aflc-adel/video/en/{display_id}', display_id)
        else:
            video_data = extract_attributes(video_element_html)
        video_id = video_data['mediaId']
```

And I don't think we should be "non-fatal" here.

Reply:
Both `article` and `a` tags have the `data-id` attribute. So, directly using `extract_attributes` on the `video_element_html` will not work as the `article` element is fetched. However, I've tried to follow what you have suggested and replaced `_search_regex` with `_search_json`.
![Screenshot from 2024-09-20 02-05-25](https://github.com/user-attachments/assets/eebba988-00b4-464b-83f1-655b84f0ac7b)